From 8b5384c834a11f97cabe4ca270795d4502730d57 Mon Sep 17 00:00:00 2001
From: dann frazier <dann.frazier@chainguard.dev>
Date: Tue, 15 Sep 2026 11:33:33 -0600
Subject: [PATCH 16/18] Revert "env,printenv: with --null, avoid CRLF
 translation on windows"

This reverts commit 50842004dc845cbc48f6caa45bd9a1371f0ded96.
NEWS left untouched.
---
 src/env.c      | 4 ----
 src/printenv.c | 4 ----
 2 files changed, 8 deletions(-)

diff --git a/src/env.c b/src/env.c
index 9c7aef854..3433db83b 100644
--- a/src/env.c
+++ b/src/env.c
@@ -29,7 +29,6 @@
 #include "printenv.h"
 #include "quote.h"
 #include "sig2str.h"
-#include "xbinary-io.h"
 
 /* The official name of this program (e.g., no 'g' prefix).  */
 #define PROGRAM_NAME "env"
@@ -908,9 +907,6 @@ main (int argc, char **argv)
     {
       char const terminator = opt_nul_terminate_output ? '\0' : '\n';
 
-      if (opt_nul_terminate_output)
-        xset_binary_mode (STDOUT_FILENO, O_BINARY);
-
       /* Print the environment and exit.  */
       for (char *const *e = environ; *e; ++e)
         print_envvar (*e, terminator, quote_output);
diff --git a/src/printenv.c b/src/printenv.c
index 540f56334..fbe35d5a6 100644
--- a/src/printenv.c
+++ b/src/printenv.c
@@ -35,7 +35,6 @@
 #include "argmatch.h"  /* argmatch($QUOTING_STYLE).  */
 #include "system.h"
 #include "printenv.h"
-#include "xbinary-io.h"
 
 /* Exit status for syntax errors, etc.  */
 enum { PRINTENV_FAILURE = 2 };
@@ -127,9 +126,6 @@ main (int argc, char **argv)
   bool ok;
   char const terminator = opt_nul_terminate_output ? '\0' : '\n';
 
-  if (opt_nul_terminate_output)
-    xset_binary_mode (STDOUT_FILENO, O_BINARY);
-
   if (optind >= argc)
     {
       for (char **env = environ; *env != NULL; ++env)
-- 
2.53.0

